pkg/lighthouse/ocp/ensure.go

```diff
@@ -0,0 +1,59 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ocp
+
+import (
+	"github.com/pkg/errors"
+	"github.com/submariner-io/subctl/pkg/clusterrole"
+	"github.com/submariner-io/subctl/pkg/clusterrolebinding"
+	operator_ocp "github.com/submariner-io/subctl/pkg/operator/ocp"
+	"github.com/submariner-io/submariner-operator/pkg/embeddedyamls"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+func EnsureRBAC(dynClient dynamic.Interface, kubeClient kubernetes.Interface, namespace string) (bool, error) {
+	if !operator_ocp.IsOcpPlatform(dynClient) {
+		return false, nil
+	}
+
+	createdAgentCR, err := clusterrole.EnsureFromYAML(kubeClient, embeddedyamls.Config_rbac_lighthouse_agent_ocp_cluster_role_yaml)
+	if err != nil {
+		return false, errors.Wrap(err, "error provisioning the Agent OCP ClusterRole resource")
+	}
+
+	createdAgentCRB, err := clusterrolebinding.EnsureFromYAML(kubeClient, namespace,
+		embeddedyamls.Config_rbac_lighthouse_agent_ocp_cluster_role_binding_yaml)
+	if err != nil {
+		return false, errors.Wrap(err, "error provisioning the Agent OCP ClusterRoleBinding resource")
+	}
+
+	createdCoreDNSCR, err := clusterrole.EnsureFromYAML(kubeClient, embeddedyamls.Config_rbac_lighthouse_coredns_ocp_cluster_role_yaml)
+	if err != nil {
+		return false, errors.Wrap(err, "error provisioning the CoreDNS OCP ClusterRole resource")
+	}
+
+	createdCoreDNSCRB, err := clusterrolebinding.EnsureFromYAML(kubeClient, namespace,
+		embeddedyamls.Config_rbac_lighthouse_coredns_cluster_role_binding_yaml)
+	if err != nil {
+		return false, errors.Wrap(err, "error provisioning the CoreDNS OCP ClusterRoleBinding resource")
+	}
+
+	return createdAgentCR || createdAgentCRB || createdCoreDNSCR || createdCoreDNSCRB, errors.Wrap(err, "error updating the SCC resources")
```

Comment:
`err` will be nil here since it's checked on L54.

```suggestion
	return createdAgentCR || createdAgentCRB || createdCoreDNSCR || createdCoreDNSCRB, nil
```